apm-sniffer/apm-sdk-plugin/spring-plugins/spring-webflux-5.x-webclient-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/webclient/WebFluxWebClientInterceptor.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import reactor.core.publisher.Mono;
+
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.util.function.BiConsumer;
+
+public class WebFluxWebClientInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        if (allArguments[0] == null) {
+            //illegal args,can't trace ignore
+            return;
+        }
+
+        ClientRequest request = (ClientRequest) allArguments[0];
+        final ContextCarrier contextCarrier = new ContextCarrier();
+
+        URI uri = request.url();
+        final String requestURIString = getRequestURIString(uri);
+        final String operationName = requestURIString;
+        final String remotePeer = getIPAndPort(uri);
+        AbstractSpan span = ContextManager.createExitSpan(operationName, contextCarrier, remotePeer);
+
+        //set componet name
+        span.setComponent(ComponentsDefine.SPRING_WEBCLIENT);
+        Tags.URL.set(span, uri.toString());
+        Tags.HTTP.METHOD.set(span, request.method().toString());
+        SpanLayer.asHttp(span);
+
+        //user async interface
+        span.prepareForAsync();
+        ContextManager.stopSpan();
+
+        objInst.setSkyWalkingDynamicField(span);
+
+        CarrierItem next = contextCarrier.items();
+        while (next.hasNext()) {
+            next = next.next();
+            request.headers().add(next.getHeadKey(), next.getHeadValue());
+        }
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        Mono<ClientResponse> ret1 = (Mono<ClientResponse>) ret;
+        AbstractSpan span = (AbstractSpan) objInst.getSkyWalkingDynamicField();
+        return ret1.doAfterSuccessOrError(new BiConsumer<ClientResponse, Throwable>() {
+            @Override
+            public void accept(ClientResponse clientResponse, Throwable throwable) {
+                HttpStatus httpStatus = clientResponse.statusCode();
+                if (httpStatus != null) {
+                    Tags.STATUS_CODE.set(span, Integer.toString(httpStatus.value()));
+                    if (httpStatus.isError()) {
+                        span.errorOccurred();
+                    }
+                }
+            }
+        }).doOnError(error -> {
+            span.errorOccurred();
+            span.log(error);
```

Comment:
You don't need `span.errorOccurred();` now
```suggestion
            span.log(error);
```